smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -179,35 +188,36 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         require(sigs.length == claims.length, "INV_CLM_LEN");
         require(sigs.length == signatureData.length, "INV_SIG_LEN");
 
+        uint256 intermediateNonce = lastNonceSubmitted;
+        require(intermediateNonce + 1 == claims[0].nonce, "INV_ORD");
+
         for (uint256 i = 0; i < sigs.length; i++) {
+            intermediateNonce++;
```

Comment:
Same comment as `submitProphecyClaimAggregatedSigs`, let's enforce ordering in this loop by checking claim data's nonce is equal to the intermediate nonce. If it isn't correct, revert.